src/socket.rs

```diff
@@ -907,6 +907,21 @@ impl UtpSocket {
         Ok(())
     }
 
+    /// Send a keepalive packet on the stream.
+    pub fn send_keepalive(&self) {
+        if now_microseconds() - self.last_acked_timestamp > 14_000_000 {
```

Comment:
We need to be sending keepalive packets so that the connections don't die when they're not in use. The `send_keepalive()` method first checks whether any packets we sent in the last 14 seconds have been acked so that we don't send a keepalive if we don't need to. 14 seconds was chosen as a number that's a bit less than half of 30 seconds. The libutp code suggests that keepalives need to be sent at least every 30 seconds (if there's no other traffic). They use a threshold of 29 seconds to check whether to send the keepalive but they also require that their send_keepalive method gets called every 500ms to ensure that the keepalive always gets sent before the 30 second mark. I figured lowering the threshold to 14s gives the user a bit more freedom to call the method less often - they only need to call it at a rate of once every 14 seconds or so. Although in our usage we call send_keepalive every 50ms so lowering the threshold doesn't give us any benefits. I dunno if my logic for setting it to 14 seconds really makes sense but it hardly makes a difference anyway.